### PR TITLE
Add column level filter type option

### PR DESCRIPTION
**Pull request by karthick-vinod**

### CODE REVIEW

File: src/components/TableFilter.js [hunk 1]
```diff
@@ -310,13 +308,20 @@ class TableFilter extends React.Component {
...
-        {options.filterType === 'checkbox'
-          ? this.renderCheckbox(columns)
-          : options.filterType === 'multiselect'
-          ? this.renderMultiselect(columns)
-          : options.filterType === 'textField'
-          ? this.renderTextField(columns)
-          : this.renderSelect(columns)}
+        <GridList cellHeight="auto" cols={2}>
+          {columns.map((column, index) => {
+            if (column.filter) {
```
Comment (guidouil) [inline src/components/TableFilter.js]: Since you copied my `if` here you should remove all ternaries doing the same verification in each sub renders 😉

Comment (karthick-vinod) [reply src/components/TableFilter.js]: I'm sorry, I didn't get your point here.
BTW, I saw your PR only after I started implementing it. So raised a PR anyway. Hope it's fine😃

Comment (guidouil) [reply src/components/TableFilter.js]: at this line if the filter boolean in true or false to show the filter and you check it again at lines 197 , 229 and 251.
Aside of that I think your integration with the `Grid` is much better than what I did 👏 
Also I copied your testing part to make my built pass and I want to thank you for that 😄

Comment (karthick-vinod) [reply src/components/TableFilter.js]: Fixed it. Thanks for pointing it out.

File: src/components/TableFilter.js [hunk 2]
```diff
@@ -148,138 +143,128 @@ class TableFilter extends React.Component {
...
-                    classes={{
-                      root: classes.checkbox,
-                      checked: classes.checked,
-                    }}
-                    value={filterColumn != null ? filterColumn.toString() : ''}
-                  />
-                }
-                label={filterColumn}
-              />
+              <Grid item>
```
Comment (guidouil) [inline src/components/TableFilter.js]: Here at line 159 the console is saying 
```
Warning: Failed prop type: The property `xs` of `Grid` must be used on `item` ...
```
Can you change this markup to `<Grid item xs>` please

File: src/MUIDataTable.js [hunk 3]
```diff
@@ -80,6 +80,7 @@ class MUIDataTable extends React.Component {
             viewColumns: PropTypes.bool,
             filterList: PropTypes.array,
             filterOptions: PropTypes.array,
+            filterType: PropTypes.oneOf(['dropdown', 'checkbox', 'multiselect', 'textField']),
```
Comment (guidouil) [inline src/MUIDataTable.js]: The filters will not filter properly unless you change line 481 of this file with this (or something similar)
```
      const { caseSensitive } = this.options;
      let { filterType } = this.options;
      if (columns[index].options && columns[index].options.filterType) {
        filterType = columns[index].options.filterType;
      }
```
I tried with a `filterType: "textField"` on the age to test this out 😉

File: src/components/TableFilter.js [hunk 4]
```diff
@@ -148,138 +143,128 @@ class TableFilter extends React.Component {
...
-              </Select>
-            </FormControl>
-          ) : (
-            false
-          ),
-        )}
-      </div>
+      <GridListTile key={index} cols={1}>
+        <div className={classes.selectRoot}>
+          <FormControl className={classes.selectFormControl} key={index}>
```
Comment (guidouil) [inline src/components/TableFilter.js]: Also some of this `key={index}` are redundant and React complains about it 
```
Warning: Each child in a list should have a unique "key" prop.
```

File: src/components/TableFilter.js (hunk 2 above)
Comment (karthick-vinod) [reply src/components/TableFilter.js]: Done

File: src/components/TableFilter.js (hunk 4 above)
Comment (karthick-vinod) [reply src/components/TableFilter.js]: Fixed it.

File: src/MUIDataTable.js (hunk 3 above)
Comment (karthick-vinod) [reply src/MUIDataTable.js]: Fixed it.

**Comment by guidouil** [comment]
@gregnb can we get your review on this PR ?
Please 🙏

**Comment by gregnb** [comment]
Hey everyone sorry for the delay. Returning from a long vacation. Allow me some time to review this PR. I would like this feature but I need to review and make appropriate changes as well.

Thank you for your patience.

**Comment by gregnb** [comment]
@karthick-vinod Thanks for your patience and this PR. It is approved and I will merge it in before I do another release later today

**Comment by PrashanthSubramanianMS** [comment]
Is this released? Also, can we add a note in the API documentation that this is not implemented? I spend two hours debugging and creating a repro before I found out this is not implemented. 

Repro: https://codesandbox.io/s/4rr332wm77

**Pull request merged**